test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,252 @@
+		o.Expect(err).ToNot(o.HaveOccurred())
+		newComMatrix, err := newComMatrixCreator.CreateEndpointMatrix()
+		o.Expect(err).ToNot(o.HaveOccurred())
+		newComMatrix.WriteMatrixToFileByType(utilsHelpers, "new-commatrix", types.FormatCSV, deployment, artifactsDir)
+
+		g.By("get documented commatrix")
+		fp := filepath.Join(artifactsDir, "doc-commatrix.csv")
+		createCSVFromUrl(docCommatrixVersionedUrl, fp)
+		docComMatrixCreator, err := commatrixcreator.New(epExporter, fp, types.FormatCSV, env, deployment)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		docComDetailsList, err := docComMatrixCreator.GetComDetailsListFromFile()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("Filter documented commatrix for diff generation")
+		// if cluster is a SNO exclude standard deployment type's static entries in diff generation
+		if isSNO {
+			// Exclude static entries of standard deployment type and all worker nodes entries.
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList, &types.ComMatrix{Matrix: types.StandardStaticEntries}, &types.ComMatrix{Matrix: docComDetailsList})
+		} else {
+			// Exclude specific master entries that are documented as both worker and master,
+			// and are open only on the worker node in standard clusters.
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList, &types.ComMatrix{Matrix: StandardExcludedMasterComDetails}, nil)
+
+		}
+
+		// if cluster is running on BM exclude Cloud static entries in diff generation
+		// else cluster is running on Cloud and exclude BM static entries in diff generation.
+		if isBM {
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList,
+				&types.ComMatrix{Matrix: types.CloudStaticEntriesMaster},
+				&types.ComMatrix{Matrix: types.CloudStaticEntriesWorker})
+		} else {
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList,
+				&types.ComMatrix{Matrix: types.BaremetalStaticEntriesMaster},
+				&types.ComMatrix{Matrix: types.BaremetalStaticEntriesWorker})
+		}
+		docComMatrix := &types.ComMatrix{Matrix: docComDetailsList}
+
+		g.By("generating diff between matrices for testing purposes")
+		diff := matrixdiff.Generate(newComMatrix, docComMatrix)
+		diffStr, err := diff.String()
+		o.Expect(err).ToNot(o.HaveOccurred())
+		err = os.WriteFile(filepath.Join(artifactsDir, "doc-diff-new"), []byte(diffFileComments+diffStr), 0644)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("comparing new and documented commatrices")
+		areEqual := comMatricesAreEqual(*newComMatrix, *docComMatrix)
+		o.Expect(areEqual).To(o.BeTrue())
+	})
+})
+
+// getClusterXYersion return cluster's version in X.Y stream
+func getClusterXYersion(configClient *configv1client.ConfigV1Client) (string, error) {
+	clusterVersion, err := configClient.ClusterVersions().Get(context.Background(), "version", metav1.GetOptions{})
+	if err != nil {
+		return "", nil
+	}
+	ClusterVersionParts := strings.SplitN(clusterVersion.Status.Desired.Version, ".", 3)
```

Comment:
Is that var with a capital letter by mistake?

Reply:
yep, thanks!